src/generate/register.rs

```diff
@@ -679,12 +697,23 @@ pub fn fields(
                         }
                     }
                 } else {
-                    quote! {
-                        ///Writes raw bits to the field
-                        #inline
-                        pub #unsafety fn #bits(self, value: #fty) -> &'a mut W {
-                            self.w.bits = (self.w.bits & !#hexmask) | (value as #rty & #hexmask);
-                            self.w
+                    if use_mask {
+                        quote! {
+                            ///Writes raw bits to the field
+                            #inline
+                            pub #unsafety fn #bits(self, value: #fty) -> &'a mut W {
+                                self.w.bits = (self.w.bits & !#hexmask) | (value as #rty & #hexmask);
+                                self.w
+                            }
+                        }
+                    } else {
+                        quote! {
+                            ///Writes raw bits to the field
+                            #inline
+                            pub #unsafety fn #bits(self, value: #fty) -> &'a mut W {
+                                self.w.bits |= value as #rty;
+                                self.w
+                            }
```

Comment:
Could these two be joined, instead using one quote for the fn decl, and two quotes for the assignment in an if else

eg 

```rust
let op = if use_mask {
    quote!{ self.w.bits = (self.w.bits & !#hexmask) | (value as #rty & #hexmask); }
else {
    quote!{ self.w.bits |= value as #rty; }
}
....
```

Reply:
Totally!